app/javascript/dashboard/App.vue

```diff
@@ -98,7 +98,7 @@ export default {
       mql.onchange = e => setColorTheme(e.matches);
     },
     setLocale(locale) {
-      this.$root.$i18n.locale = locale;
+      this.$root.$i18n.locale = locale.replace('-', '_');
```

Comment:
This may not be the most reliable. Just updating the `this.$i18n.locale` should work better